staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/validation.go

```diff
@@ -105,16 +108,18 @@ func validate(pth *field.Path, s *structuralschema.Structural, rootSchema *struc
 				allErrs = append(allErrs, errs...)
 			} else if errs := apiservervalidation.ValidateCustomResource(pth.Child("default"), s.Default.Object, validator); len(errs) > 0 {
 				allErrs = append(allErrs, errs...)
-			} else if celValidator := cel.NewValidator(s); celValidator != nil {
-				allErrs = append(allErrs, celValidator.Validate(pth.Child("default"), s, s.Default.Object)...)
+			} else if celValidator := cel.NewValidator(s, cel.PerCallLimit); celValidator != nil {
+				err, rmCost := celValidator.Validate(pth.Child("default"), s, s.Default.Object, remainingCost)
+				allErrs = append(allErrs, err...)
+				remainingCost -= rmCost
```

Comment:
is `-=` here correct? I thought this should be `remainingCost = rmCost`, and short-circuiting if < 0

if this is currently wrong, need a test exercising this

Reply:
I have fixed it. And will add test in https://github.com/kubernetes/kubernetes/pull/108646 when able to pass context.